### ISSUE TITLE
Track branch `api`

### ISSUE THREAD

**Issue by dbarrosop** [napalm-automation-community/napalm-ros#24]
This issue is just to discuss and keep track of the progress of the API branch.

**Comment by dbarrosop** [comment]
@luqasz are you gonna be migrating the rest of the methods to the API?

**Comment by dbarrosop** [comment]
Pinging @inetuid, would be cool if you could review the API branch and contribute as well : )

**Comment by luqasz** [comment]
I will migrate more methods into those using API

**Comment by dbarrosop** [comment]
Thx, great. If you need help with something let me know.

@inetuid, are you aware of this work?

**Comment by inetuid** [comment]
I'm aware. I suggest the easiest way to integrate would be to replace the current _api_get() method that calls SSH rather than go through and replace all instances of this with the proposed api() method.

It would require fewer changes to the code which would make life easier as I'm still trying to address issues in the current codebase.

**Comment by luqasz** [comment]
Shall I migrate all methods in one pull request or one by one ?

**Comment by inetuid** [comment]
I've just read #20 which contradicts my point above. My preference was for it to be all or nothing. I didn't want to get 99% of the way through converting the code to use the API to discover some last little thing that required SSH to achieve and then need to support both mechanisms.

**Comment by dbarrosop** [comment]
@inetuid 

> My preference was for it to be all or nothing

That's what we are doing. That's why all the PRs are going against the `api` branch. Once everything has been migrated we will merge into develop and then release. The reason to do it like this is so we can work in parallel, collaborate and have smaller PRs. But we agree with you, we want to use the API for everything.

**Comment by dbarrosop** [comment]
@inetuid I understand there might be some confussion here because conversation has grown somehow organically across different PRs. Sorry about that. The relevant comment that agrees with your approach is this:

https://github.com/napalm-automation/napalm-ros/pull/22#issuecomment-262320614

**Comment by dbarrosop** [comment]
@luqasz do as you prefer. It also depends on how long you think it's gonna take for you to migrate everything, if we are talking weeks I'd suggest sending a PR every time you get a couple of methods migrated.

**Comment by inetuid** [comment]
I suggest converting the methods that call self.cli() (and the traceroute() method) first as these are the ones I have either not tested using the API or didn't know how to do via the API. The methods that currently call self._api_get() should already work (or at least I've tested against my private API branch.

**Comment by luqasz** [comment]
@inetuid Where did you get sample data for eg. `test/unit/mocked_data/test_get_interfaces_counters/normal/_interface_ethernet_print_without_paging_stats_where_name___ether1_` ? I'd like to launch a mikrotik in virtualbox and test against it. And which version also.

**Comment by inetuid** [comment]
Off the top of my head, it came from my home router running 6.3X something. Do you not get it in Virtualbox?

**Comment by luqasz** [comment]
I do have working virtualbox setup but I do not have all things such as `/system/health` voltage etc.

**Comment by inetuid** [comment]
Yeah, there are some physical things (fans, power, switch chips, optics) that you can't replicate in the virtual ROS instance. You need access to a physical device for these.
